h5py/tests/test_file.py

```diff
@@ -673,7 +673,8 @@ def test_unicode_hdf5_python_consistent(self):
         """
         fname = self.mktemp(prefix=chr(0x201a))
         with File(fname, 'w') as f:
-            self.assertTrue(os.path.exists(fname))
+            print(os.listdir(self.tempdir))
+            assert os.path.exists(fname)
```

Comment:
```suggestion
        assert os.path.exists(fname)
```

maybe check after we have closed the file?